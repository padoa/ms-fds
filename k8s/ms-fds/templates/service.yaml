apiVersion: v1
kind: Service
metadata:
  name: {{ include "ms-fds.fullname" . }}
  labels:
    {{- include "ms-fds.labels" . | nindent 4 }}
    app: ms-fds
    service: ms-fds
    service.istio.io/canonical-name: ms-fds
spec:
  type: {{ .Values.service.type }}
  ports:
    - port: {{ .Values.service.port }}
      targetPort: http
      protocol: TCP
      name: http
  selector:
    {{- include "ms-fds.selectorLabels" . | nindent 4 }}
