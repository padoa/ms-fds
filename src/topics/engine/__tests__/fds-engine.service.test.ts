import type { SpyInstance } from 'vitest';
import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest';

import { FdsEngineService } from '@topics/engine/fds-engine.service.js';
import { FDS_TEST_FILES_PATH } from '@src/__fixtures__/fixtures.constants.js';
import { ExtractionRulesService } from '@topics/engine/rules/extraction-rules.service.js';
import { FdsTreeCleanerService } from '@topics/engine/transformer/fds-tree-cleaner.service.js';
import { FdsTreeBuilderService } from '@topics/engine/transformer/fds-tree-builder.service.js';
import { PdfParserService } from '@topics/engine/pdf-parser/pdf-parser.service.js';

describe('FdsEngine tests', () => {
  describe('ExtractDataFromFds tests', () => {
    it('should extract data from a pdf Fds file', async () => {
      await expect(FdsEngineService.extractDataFromFds(FDS_TEST_FILES_PATH.PDF_JEFFACLEAN)).resolves.toMatchSnapshot();
    });

    describe('Error cases', () => {
      let parsePDFSpy: SpyInstance;
      let buildFdsTreeSpy: SpyInstance;
      let cleanFdsTreeSpy: SpyInstance;
      let extractSpy: SpyInstance;

      beforeEach(() => {
        parsePDFSpy = vi.spyOn(PdfParserService, 'parsePDF');
        buildFdsTreeSpy = vi.spyOn(FdsTreeBuilderService, 'buildFdsTree');
        cleanFdsTreeSpy = vi.spyOn(FdsTreeCleanerService, 'cleanFdsTree');
        extractSpy = vi.spyOn(ExtractionRulesService, 'extract');
      });

      afterEach(() => {
        buildFdsTreeSpy.mockRestore();
        parsePDFSpy.mockRestore();
        cleanFdsTreeSpy.mockRestore();
        extractSpy.mockRestore();
      });

      it('should handle error when the parsing crashes', async () => {
        parsePDFSpy.mockImplementation(() => {
          throw new Error();
        });
        expect(await FdsEngineService.extractDataFromFds(FDS_TEST_FILES_PATH.PDF_JEFFACLEAN)).toEqual({ dataExtracted: null, fromImage: null });
      });

      it('should handle error when the tree builder crashes', async () => {
        buildFdsTreeSpy.mockImplementation(() => {
          throw new Error();
        });
        expect(await FdsEngineService.extractDataFromFds(FDS_TEST_FILES_PATH.PDF_JEFFACLEAN)).toEqual({ dataExtracted: null, fromImage: null });
      });

      it('should handle error when the cleaning crashes', async () => {
        cleanFdsTreeSpy.mockImplementation(() => {
          throw new Error();
        });
        expect(await FdsEngineService.extractDataFromFds(FDS_TEST_FILES_PATH.PDF_JEFFACLEAN)).toEqual({ dataExtracted: null, fromImage: null });
      });

      it('should handle error when the extraction crashes', async () => {
        extractSpy.mockImplementation(() => {
          throw new Error();
        });
        expect(await FdsEngineService.extractDataFromFds(FDS_TEST_FILES_PATH.PDF_JEFFACLEAN)).toEqual({ dataExtracted: null, fromImage: null });
      });
    });

    it('should extract data from an image Fds file', async () => {
      await expect(FdsEngineService.extractDataFromFds(FDS_TEST_FILES_PATH.IMAGE_DEGRAISSANT)).resolves.toMatchSnapshot();
    });
  }, 70000);
});
