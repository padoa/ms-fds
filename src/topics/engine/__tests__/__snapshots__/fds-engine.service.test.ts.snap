// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FdsEngine tests > ExtractDataFromFds tests > should extract data from a pdf Fds file 1`] = `
{
  "dataExtracted": {
    "dangers": [
      {
        "code": "h304",
        "metaData": {
          "endBox": undefined,
          "startBox": {
            "pageNumber": 1,
            "xPositionProportion": 0.09003440120404214,
            "yPositionProportion": 0.7744963892056252,
          },
        },
      },
      {
        "code": "p301 + p 310",
        "metaData": {
          "endBox": undefined,
          "startBox": {
            "pageNumber": 2,
            "xPositionProportion": 0.0769189421629757,
            "yPositionProportion": 0.2658494868871152,
          },
        },
      },
      {
        "code": "p331",
        "metaData": {
          "endBox": undefined,
          "startBox": {
            "pageNumber": 2,
            "xPositionProportion": 0.3967157600516018,
            "yPositionProportion": 0.30693652603572785,
          },
        },
      },
      {
        "code": "p 501",
        "metaData": {
          "endBox": undefined,
          "startBox": {
            "pageNumber": 2,
            "xPositionProportion": 0.3967157600516018,
            "yPositionProportion": 0.3341505131128848,
          },
        },
      },
    ],
    "date": {
      "formattedDate": "2017/01/20",
      "inTextDate": "20 janvier 2017",
    },
    "physicalState": {
      "metaData": {
        "endBox": undefined,
        "startBox": {
          "pageNumber": 7,
          "xPositionProportion": 0.12333369167920878,
          "yPositionProportion": 0.2668377042949449,
        },
      },
      "value": "liquide",
    },
    "producer": {
      "metaData": {
        "endBox": undefined,
        "startBox": {
          "pageNumber": 1,
          "xPositionProportion": 0.08097720920232208,
          "yPositionProportion": 0.3348726719878373,
        },
      },
      "name": "jeffa lubrifiants",
    },
    "product": {
      "metaData": {
        "endBox": undefined,
        "startBox": {
          "pageNumber": 1,
          "xPositionProportion": 0.08097720920232208,
          "yPositionProportion": 0.2719688331432915,
        },
      },
      "name": "jeffaclean h 100 sc",
    },
    "substances": [
      {
        "casNumber": "64742-52-5",
        "ceNumber": "265-155-0",
      },
      {
        "casNumber": "67762-38-3",
        "ceNumber": "267-015-4",
      },
      {
        "casNumber": undefined,
        "ceNumber": "918-481-9",
      },
    ],
  },
  "fromImage": false,
}
`;

exports[`FdsEngine tests > ExtractDataFromFds tests > should extract data from an image Fds file 1`] = `
{
  "dataExtracted": {
    "dangers": [
      {
        "code": "h314",
        "metaData": {
          "endBox": {
            "pageNumber": 2,
            "xPositionProportion": 0.7142857142857143,
            "yPositionProportion": 0.8848484848484849,
          },
          "startBox": {
            "pageNumber": 2,
            "xPositionProportion": 0.20095238095238097,
            "yPositionProportion": 0.8740740740740741,
          },
        },
      },
      {
        "code": "h412",
        "metaData": {
          "endBox": {
            "pageNumber": 2,
            "xPositionProportion": 0.7685714285714286,
            "yPositionProportion": 0.898989898989899,
          },
          "startBox": {
            "pageNumber": 2,
            "xPositionProportion": 0.20095238095238097,
            "yPositionProportion": 0.8882154882154882,
          },
        },
      },
      {
        "code": "p273",
        "metaData": {
          "endBox": {
            "pageNumber": 3,
            "xPositionProportion": 0.47523809523809524,
            "yPositionProportion": 0.3111111111111111,
          },
          "startBox": {
            "pageNumber": 3,
            "xPositionProportion": 0.2019047619047619,
            "yPositionProportion": 0.2996632996632997,
          },
        },
      },
      {
        "code": "p280",
        "metaData": {
          "endBox": {
            "pageNumber": 3,
            "xPositionProportion": 0.8495238095238096,
            "yPositionProportion": 0.32525252525252524,
          },
          "startBox": {
            "pageNumber": 3,
            "xPositionProportion": 0.2019047619047619,
            "yPositionProportion": 0.31447811447811447,
          },
        },
      },
      {
        "code": "p301 + p330 + p331",
        "metaData": {
          "endBox": {
            "pageNumber": 3,
            "xPositionProportion": 0.7485714285714286,
            "yPositionProportion": 0.35084175084175084,
          },
          "startBox": {
            "pageNumber": 3,
            "xPositionProportion": 0.2019047619047619,
            "yPositionProportion": 0.34276094276094277,
          },
        },
      },
      {
        "code": "p303 + p361 + p353",
        "metaData": {
          "endBox": {
            "pageNumber": 3,
            "xPositionProportion": 0.8495238095238096,
            "yPositionProportion": 0.36632996632996634,
          },
          "startBox": {
            "pageNumber": 3,
            "xPositionProportion": 0.2019047619047619,
            "yPositionProportion": 0.35622895622895623,
          },
        },
      },
      {
        "code": "p304 + p340",
        "metaData": {
          "endBox": {
            "pageNumber": 3,
            "xPositionProportion": 0.8504761904761905,
            "yPositionProportion": 0.3939393939393939,
          },
          "startBox": {
            "pageNumber": 3,
            "xPositionProportion": 0.2019047619047619,
            "yPositionProportion": 0.3838383838383838,
          },
        },
      },
      {
        "code": "p305 + p351 + p338",
        "metaData": {
          "endBox": {
            "pageNumber": 3,
            "xPositionProportion": 0.8504761904761905,
            "yPositionProportion": 0.4208754208754209,
          },
          "startBox": {
            "pageNumber": 3,
            "xPositionProportion": 0.2019047619047619,
            "yPositionProportion": 0.41144781144781145,
          },
        },
      },
      {
        "code": "p310",
        "metaData": {
          "endBox": {
            "pageNumber": 3,
            "xPositionProportion": 0.680952380952381,
            "yPositionProportion": 0.46195286195286195,
          },
          "startBox": {
            "pageNumber": 3,
            "xPositionProportion": 0.2019047619047619,
            "yPositionProportion": 0.45252525252525255,
          },
        },
      },
      {
        "code": "p501",
        "metaData": {
          "endBox": {
            "pageNumber": 3,
            "xPositionProportion": 0.8504761904761905,
            "yPositionProportion": 0.47609427609427607,
          },
          "startBox": {
            "pageNumber": 3,
            "xPositionProportion": 0.2019047619047619,
            "yPositionProportion": 0.46464646464646464,
          },
        },
      },
    ],
    "date": {
      "formattedDate": "2016/09/09",
      "inTextDate": "09/09/16",
    },
    "physicalState": null,
    "producer": {
      "metaData": {
        "endBox": {
          "pageNumber": 1,
          "xPositionProportion": 0.47619047619047616,
          "yPositionProportion": 0.5858585858585859,
        },
        "startBox": {
          "pageNumber": 1,
          "xPositionProportion": 0.37333333333333335,
          "yPositionProportion": 0.5764309764309764,
        },
      },
      "name": "hypred sas",
    },
    "product": null,
    "substances": [
      {
        "casNumber": "69011-36-5",
        "ceNumber": undefined,
      },
      {
        "casNumber": "08062-26-4",
        "ceNumber": undefined,
      },
      {
        "casNumber": undefined,
        "ceNumber": "219-145-8",
      },
    ],
  },
  "fromImage": true,
}
`;
